### PR TITLE
feat(app): implementation of filters for questions and multi choices

### PR DESCRIPTION
**Pull request by axelgcd**
Changes : 
- changed used DB
- added const filters to get the player names used as multi choices
- changed sendQuestionToFrontend to get the player has to guess depending on his tier 

More info : 
- setTier, acessDB, updateTierAndScore were code already pushed thqt might contain usefull stuff and I need to dive deeper on it but I would like to work on it on the next pr

### CODE REVIEW

File: controllers/data.controller.js [hunk 1]
```diff
@@ -3,36 +3,53 @@ const AuthService = require("../services/auth.service");
 
 const { MongoClient } = require("mongodb");
 const { update } = require("./user.controller");
-const uri = "mongodb+srv://testaccount:Djsejalffl12@cluster0.s9ohawm.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0"
+const uri =
+  "mongodb+srv://developers:djsejalffl1234@undermilli.u8syqsv.mongodb.net/?retryWrites=true&w=majority&appName=Undermilli";
```
Comment (ivxvi12) [inline controllers/data.controller.js]: Can we use .env file for this one?

**Review by ivxvi12 — CHANGES_REQUESTED** [review]

File: controllers/data.controller.js [hunk 2]
```diff
@@ -43,669 +60,613 @@ const client = new MongoClient(uri);
...
-        // TODO: 이 filter로 뽑아내는 문제들이 바뀔것임. 
-        let filter;
-        if (userScore === 0) {
-            filter = {YEAR: 23};
-        } else if (userScore === 1) {
-            filter = {$and: [
-                {YEAR: {$in: [20, 22]}},
-                {LEAGUE: "LCS"}
-            ]};
+    console.log(`이게 유저 점수임 ${userScore}`);
```
Comment (aknur1999) [inline controllers/data.controller.js]: is this console.log essential?

**Comment by aknur1999** [comment]
@axelgcd can you write description what you have done? so i can know what to check, thank you!

**Comment by axelgcd** [comment]
> @axelgcd can you write description what you have done? so i can know what to check, thank you!

I added some description

File: controllers/data.controller.js [hunk 3]
```diff
@@ -43,669 +60,613 @@ const client = new MongoClient(uri);
...
+    const coll = db.collection("users");
+
+    // Get the user's ID from the request object
+    const userID = req.user._id;
+
+    // Retrieve the user's document from the database
+    const userDocument = await coll.findOne({ _id: userID });
+
+    // Check if the "tier" field exists in the user's document
+    if (userDocument.tier === " ") {
```
Comment (aknur1999) [inline controllers/data.controller.js:91]: I think you can set default value for tier as I1. if (userDocument && !userDocument.tier) then you can change it to S4. So you dont need to write await coll.updateOne twice.

File: controllers/data.controller.js [hunk 4]
```diff
@@ -43,669 +60,613 @@ const client = new MongoClient(uri);
...
-                    updatedTier = "M";
-                    updatedScore = currentScore - 100;
-                } 
-
-            }
-
+  try {
+    await client.connect();
+
+    const db = client.db("test");
```
Comment (aknur1999) [inline controllers/data.controller.js:136]: I think accessing to db and coll in every function is not efficient. Maybe you can create new model or service so you can use it in every function

File: controllers/data.controller.js [hunk 5]
```diff
@@ -43,669 +60,613 @@ const client = new MongoClient(uri);
...
+    await client.connect();
+
+    const db = client.db("sample_data");
+    const coll = db.collection("final_real_data");
+
+    // Define filter based on the user's tier
+    let filter = {};
+
+    // Find documents matching the filter
+    const cursor = coll.find(filter);
```
Comment (aknur1999) [inline controllers/data.controller.js]: if filter is empty why you need to define filter here..?

File: controllers/data.controller.js [hunk 6]
```diff
@@ -756,495 +715,515 @@ exports.getMultipleChoiceAnswers = async (req, res) => {
...
-    else if (["M", "GM", "C"].includes(currentTier)) {
-        filter = {
-            $and: [
-                { YEAR: selectedQuestion.YEAR, LEAGUE: selectedQuestion.LEAGUE, POS: selectedQuestion.POS },
-                { IGN: { $ne: selectedQuestion.IGN } } // Exclude the selected document's IGN
-            ]
-        };
+exports.sendQuestionToFrontend = async (req, res) => {
+  try {
+    await client.connect();
```
Comment (aknur1999) [inline controllers/data.controller.js:469]: same in all functions, please utilize connection with client, db, and coll

File: controllers/data.controller.js [hunk 7]
```diff
@@ -756,495 +715,515 @@ exports.getMultipleChoiceAnswers = async (req, res) => {
...
+    const db = client.db("yskm");
+    const coll = db.collection("userdb");
+    const currentTier = req.user.tier;
+    console.log("currentTier:", currentTier);
```
Comment (aknur1999) [inline controllers/data.controller.js]: remove unnecessary console.logs

**Review by aknur1999 — CHANGES_REQUESTED** [review]

**Comment by aknur1999** [comment]
In addition, I think you can remove unnecessary comments for good code readability

**Review by aknur1999 — APPROVED** [review]

**Review by ivxvi12 — APPROVED** [review]
I think we have to check other things in future development.

**Pull request merged**